### PR TITLE
MNT: expire `ContourSet` deprecations

### PR DESCRIPTION
**Pull request by rcomer**
## PR summary

Expire the v3.8 deprecations for `ContourSet`.  ~The two methods removed from `ContourLabeller` seem to have been really intended for internal use and I couldn't find a v3.8 changenote for them, so I didn't add a changenote for them here either.~

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [n/a] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -406,97 +406,6 @@ def interp_vec(x, xp, fp): return [np.interp(x, xp, col) for col in fp.T]
 
         return angle, Path(xys, codes)
 
-    @_api.deprecated("3.8")
-    def calc_label_rot_and_inline(self, slc, ind, lw, lc=None, spacing=5):
```
Comment (QuLogic) [inline lib/matplotlib/contour.py:410]: Not documented?

File: lib/matplotlib/contour.py [hunk 2]
```diff
@@ -519,12 +428,6 @@ def add_label(self, x, y, rotation, lev, cvalue):
         # Add label to plot here - useful for manual mode label selection
         self.axes.add_artist(t)
 
-    @_api.deprecated("3.8", alternative="add_label")
-    def add_label_clabeltext(self, x, y, rotation, lev, cvalue):
```
Comment (QuLogic) [inline lib/matplotlib/contour.py:523]: Also not documented?

File: lib/matplotlib/contour.py (hunk 1 above)
Comment (rcomer) [reply lib/matplotlib/contour.py:410]: I couldn't see a changenote for these two at v3.8 so followed that.  My impression is that, despite being technically public, these would only have been used internally.

Comment (QuLogic) [reply lib/matplotlib/contour.py:410]: That's odd; I'd still document that it was removed though.

File: doc/api/next_api_changes/removals/28767-REC.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+returns an array.
+
+``tcolors`` and ``tlinewidths`` attributes of ``ContourSet``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+... have been removed.  Use `~.Collection.get_facecolor`, `~.Collection.get_edgecolor`
+or `~.Collection.get_linewidths` instead.
+
+
+``calc_label_rot_and_inline`` method of ``ContourLabeler``
```
Comment (rcomer) [inline doc/api/next_api_changes/removals/28767-REC.rst:22]: I first did this as `ContourLabeler.calc_label_rot_and_inline`, but that rendered as an unwrapped line that was too long to show properly on my laptop screen.

File: lib/matplotlib/contour.py [hunk 4]
```diff
@@ -406,97 +406,6 @@ def interp_vec(x, xp, fp): return [np.interp(x, xp, col) for col in fp.T]
...
-        break contour if desired.
-        """
-
-        if lc is None:
-            lc = []
-        # Half the label width
-        hlw = lw / 2.0
-
-        # Check if closed and, if so, rotate contour so label is at edge
-        closed = _is_closed_polygon(slc)
```
Comment (QuLogic) [inline lib/matplotlib/contour.py:436]: It appears that this is probably unused and can be removed as well.

File: lib/matplotlib/contour.py [hunk 5]
```diff
@@ -906,57 +809,9 @@ def __init__(self, ax, *args,
...
-        self.set_antialiased(aa)
-
-    @_api.deprecated("3.8")
-    @property
-    def collections(self):
-        # On access, make oneself invisible and instead add the old-style collections
-        # (one PathCollection per level).  We do not try to further split contours into
-        # connected components as we already lost track of what pairs of contours need
-        # to be considered as single units to draw filled regions with holes.
-        if not hasattr(self, "_old_style_split_collections"):
```
Comment (QuLogic) [inline lib/matplotlib/contour.py:933]: There's another check for `_old_style_split_collections` in `_split_path_and_get_label_rotation` that can be removed.

**Review by QuLogic — APPROVED** [review]

**Review by ksunden — APPROVED** [review]

**Pull request merged**